service/controller/v22/cluster_resource_set.go

```diff
@@ -478,8 +493,22 @@ func NewClusterResourceSet(config ClusterResourceSetConfig) (*controller.Resourc
 		}
 	}
 
+	var workerASGNameResource controller.Resource
+	{
+		c := workerasgname.ResourceConfig{
+			Logger: config.Logger,
+		}
+
+		workerASGNameResource, err = workerasgname.NewResource(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
 	resources := []controller.Resource{
 		accountIDResource,
+		workerASGNameResource,
+		asgStatusResource,
```

Comment:
This is still executed before `cloudformation` resource which kind of makes this to be one reconciliation loop behind the stack state, but after reading this PR I think this is required in order to have ASG settings available for determining if an update to cloudformation stack is needed?

Reply:
Right, we need to fetch the ASG props before to know how to deal with the update. Then, the worker ASG name is still not accessible when the CF stack is transitioning. I discussed this with Timo and the rest is optimization, since the situation improved already.